### DISCUSSION TITLE
Custom templates and interactive

### DISCUSSION THREAD

**Discussion by rukshanfonseka** [walt-id/.github#56] [Help] [answered]
I have imported a custom template to the SSI kit:
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1"
  ],
  "id": "",
  "type": ["VerifiableCredential"],
  "issuer": "did:example:issuer",
  "issuanceDate": "2022-01-01T12:34:56Z",
  "credentialSubject": {
    "id": "did:example:subject",
    "name": "John Doe",
    "age": 30
  }
}

Then when I go to issue this VC using the SSIKit and --interactive I was expecting to be prompted for the name and age but it simply creates the VC with the default values.

1) How do I get SSI kit to prompt for name and age when using in interactive mode?
2) What command line can I use in the SSIKit to specify as part of the arguments the values of name and age?

**Comment by mikeplotean** [comment, marked as answer]
Hi @rukshanfonseka 
_Interactive_ mode is intended for demo purposes only, thus it can be used only with the pre-built templates. In this case, you're using a custom template, so it won't work. You'll have to implement a data provider for that custom credential.

A solution that might work is running the ssikit with the serve command, this will start a server locally and expose several endpoints on _localhost_.

You can use the _credentials/issue_ endpoint of the **Signatory API** hosted at http://127.0.0.1:7001/ and pass the credential with the desired values:
- either using _curl_:
```
curl -X 'POST' \
  'http://127.0.0.1:7001/v1/credentials/issue' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
      "@context": [
      "https://www.w3.org/2018/credentials/v1"
      ],
      "id": "",
      "type": ["VerifiableCredential"],
      "issuer": "did:example:issuer",
      "issuanceDate": "2022-01-01T12:34:56Z",
      "credentialSubject": {
      "id": "did:example:subject",
      "name": "<desired-name>",
      "age": <desired-age>
      }
    }'
```
- or _swagger_: http://127.0.0.1:7001/v1/swagger#/Credentials/issue

Check the [Signatory-API documentation](https://docs.walt.id/v/ssikit/getting-started/rest-apis/signatory-api#credentials) for more details.

**Reply by atharva-pb** [reply]
Hi @mikeplotean 

I was facing this too and came across this response. Thanks. However, I had a doubt:

I want to use just the SSIKIT CLI for importing a custom template and adding my value to a parameter in that template, how do I go about it? I successfully imported but am unable to enter a parameter to the custom template's fields. Could you help me with this? Or, are REST API the only way to move ahead with custom template issuance and verification for now?

Thanks,
Atharva
